Refactor

vmarsi opened the pull request.

Commit:
-bug fixed in FloodWaveDetector's run()
-modules added

Commit:
add StatisticalAnalysis class

Commit:
find_vertices now contains dictionaries instead of lists

Commit:
nullpontok_fontos.json will be downloaded by dataloader

Commit:
calculating slopes while saving the edges

Commit:
docstrings added

Commit:
docstrings added

Commit:
added median and std to the dataframe

Commit:
bug fixed

zsvizi commented on src/statistical_analysis.py:
```diff
@@ -0,0 +1,229 @@
+import json
```
Since we have az `Analysis` class, it is not clear what is the scope of this class. I am quite sure, that we need (at least) two classes, but I do not see a clear separation between them.

vmarsi replied:
Analysis class was renamed to GraphAnalysis and any function that needs a graph as an input is now in this class. Functions that create graphs (yearly) or calculate statistics simply by reading files are in StatisticalAnalysis.

vmarsi replied:
Update: I needed an AnalysisHandler class too

zsvizi commented on src/flood_wave_core/flood_wave_handler.py:
```diff
@@ -63,19 +64,82 @@ def find_dates_for_next_gauge(
+
+        distance = float(next_river_km - current_river_km)
+
+        return current_vertices, next_vertices, current_null, next_null, distance
```
Having 5 return values is a very bad practice. Moreover as I see in `FloodWaveDetector.find_edges`, these return values are used only in `FloodWaveHandler.get_slopes`, which gives the idea to create a class for slope calculation, where these return values can be member variables. I would instantiate a new `SlopeCalculator` class at every iteration of the `for` loop. This will lead to clearer calls in `find_edges` keeping its readability.

zsvizi commented on src/flood_wave_core/flood_wave_detector.py:
```diff
@@ -9,12 +9,12 @@
 import pandas as pd
```
The name of this files folder could be only `core`.

zsvizi commented on src/filtering/filtering.py:
```diff
@@ -1,6 +1,6 @@
 import networkx as nx
 
-from src.filtering_handler import FilteringHandler
+from src.filtering.filtering_handler import FilteringHandler
```
The word filtering can be misleading in a code for processing time series since there are "filters" in signal processing and stochastics for this object. I would rename everything to "select" (then instead of filtering, use selection).

zsvizi commented:
There is a block of code in `flood_wave_detector.py` in LOC88-95. I would create a method for that part in the class with an expressive name and return value `existing_gauges`.

zsvizi commented:
I have a feeling, that we could create a simple method for saving in the steps of flood wave detection (LOC142-153, LOC230-240), with only 1-2 function arguments. Could you take a look at this?

Commit:
function for calculating the ratio of the high water level vertices and all vertices

Commit:
graph analysis and statistical analysis function are separated

Commit:
some files and functions were renamed

Commit:
class SlopeCalculator was created

Commit:
existing_gauges function was created

Commit:
saving is now done in a function

Commit:
- new class created for the utilities
- branching of the graph can now be found
- calculating velocities is now possible with branching

Commit:
"cleaned" flood waves can be returned now

Commit:
docstrings added

zsvizi committed:
Rename existing_gauges method
